server/queues/processors/NotificationsProcessor.ts

```diff
@@ -237,6 +247,23 @@ export default class NotificationsProcessor extends BaseProcessor {
       return false;
     }
 
+    // Deliver only a single notification in a 12 hour window
+    const notification = await Notification.findOne({
+      order: [["createdAt", "DESC"]],
+      where: {
+        userId: user.id,
+        documentId: document.id,
+        emailedAt: {
```

Comment:
Need to make sure this has an index in the migration (see previous migrations for examples)